homeassistant/components/opnsense/__init__.py

```diff
@@ -0,0 +1,72 @@
+"""Support for OPNSense Routers."""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_URL, CONF_API_KEY, CONF_VERIFY_SSL
+from homeassistant.helpers.discovery import async_load_platform
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_API_SECRET = "api_secret"
+CONF_TRACKER_INTERFACE = "tracker_interfaces"
+
+DOMAIN = "opnsense"
+
+OPNSENSE_DATA = DOMAIN
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_URL): cv.url,
+                vol.Required(CONF_API_KEY): cv.string,
+                vol.Required(CONF_API_SECRET): cv.string,
+                vol.Optional(CONF_VERIFY_SSL, default=False): cv.boolean,
+                vol.Optional(CONF_TRACKER_INTERFACE): vol.All(
+                    cv.ensure_list, [cv.string]
+                ),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+def setup(hass, config):
+    """Set up the opnsense component."""
+    conf = config[DOMAIN]
+    url = conf[CONF_URL]
+    api_key = conf[CONF_API_KEY]
+    api_secret = conf[CONF_API_SECRET]
+    verify_ssl = conf.get(CONF_VERIFY_SSL, False)
+    tracker_interfaces = conf.get(CONF_TRACKER_INTERFACE, None)
+
+    from pyopnsense import diagnostics
+
+    if tracker_interfaces:
+        # Verify that specified tracker interfaces are valid
+        netinsight_client = diagnostics.NetworkInsightClient(
+            api_key, api_secret, url, verify_ssl
+        )
```

Comment:
I assume that there will be a traceback if the credentials are wrong or the device is not present.

Reply:
Hmm, actually it doesn't fail until it is actually used (which to be fair is the next line). The init just stores the passed in values: https://github.com/mtreinish/pyopnsense/blob/master/pyopnsense/client.py#L34-L39

Reply:
We should return False from component setup if credentials are bad.

Reply:
Fixed by: https://github.com/home-assistant/home-assistant/pull/26834/commits/3f762edbf417ac6154d9216eadbfed1193d88bf9